WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -970,7 +963,7 @@ protected Address doInBackground(Double... args) {
         }
 
         protected void onPostExecute(@Nullable Address address) {
-            if (!isAdded() || address == null || address.getMaxAddressLineIndex() == 0) {
+            if (address == null || address.getMaxAddressLineIndex() == 0) {
```

Comment:
@nbradbury I thought that too. If you look at the current code, there is always an `isAdded` check before accessing the UI elements. However, after chatting with @maxme about the whole thing, if I didn't misunderstand him, the only time we actually need to check for `isAdded` is before accessing `getActivity` since that can be `null`. Am I wrong about this?

P.S: This feels so backwards too me. When I first switched to Android, I thought `isAdded` was a way to check if the fragment is still shown and we didn't want to try & update something that's not visible. Then I thought if the fragment is not added, the UI elements (textviews and such) would be null, so it's a way to avoid NPE.

There is no documentation about this as far as I can find, but my research tells me that `isAdded` is a combination of `getActivity() == null` and actually checking if the fragment is added and most of the time it's a replacement for null checking the `getActivity`. I am still pretty confused about all this though :(